python/test/unit/fem/test_custom_assembler.py

```diff
@@ -71,27 +73,36 @@ def get_matsetvalues_cffi_api():
     else:
         raise RuntimeError("Could not find DOLFINx pkgconfig file")
 
-    cffi_support.register_type(ffi.typeof('float _Complex'), numba.types.complex64)
-    cffi_support.register_type(ffi.typeof('double _Complex'), numba.types.complex128)
+    cffi_support.register_type(ffi.typeof("float _Complex"), numba.types.complex64)
+    cffi_support.register_type(ffi.typeof("double _Complex"), numba.types.complex128)
 
-    petsc_dir = PETSc_get_config()['PETSC_DIR']
+    petsc_dir = PETSc_get_config()["PETSC_DIR"]
     petsc_arch = petsc4py.lib.getPathArchPETSc()[1]
 
-    worker = os.getenv('PYTEST_XDIST_WORKER', None)
+    worker = os.getenv("PYTEST_XDIST_WORKER", None)
     module_name = f"_petsc_cffi_{worker}"
     if MPI.COMM_WORLD.Get_rank() == 0:
         ffibuilder = cffi.FFI()
-        ffibuilder.cdef("""typedef int... PetscInt;
+        ffibuilder.cdef(
```

Comment:
Am I reading this right: ruff also formatted the C code in the string, not only the python code around it?

That is not an issue here, but we may want to be aware of this when applying ruff format e.g. to ffcx, which I guess has more parts like this one.

Reply:
In general though, I think it is bad that ruff edits the contents of triple quotes.